src/Data/Profunctor/Join.purs

```diff
@@ -0,0 +1,21 @@
+module Data.Profunctor.Join where
+
+import Prelude
+
+import Data.Functor.Invariant (class Invariant)
+import Data.Newtype (class Newtype)
+import Data.Profunctor (class Profunctor, dimap)
+
+-- | Turns a `Profunctor` into a `Invariant` functor by equating the two type
+-- | arguments.
+newtype Join p a = Join (p a a)
```

Comment:
Wait! I just realised something, this is https://github.com/purescript/purescript-monoid/pull/34 if we provide

``` purescript
instance semigroupJoin :: Semigroupoid p => Semigroup (Join p a) where
  append (Join a) (Join b) = Join (a <<< b)

instance monoidJoin :: Category p => Monoid (Join p a) where
  mempty = Join id
```